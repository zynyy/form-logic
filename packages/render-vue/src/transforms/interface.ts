import { BtnFieldsItem, LogicListItem, MetaSchema, MetaSchemaData, SchemaPattern } from '@/interface';
import { ISchema } from '@formily/json-schema';

export interface TransformsSchemaOptions {
  metaSchema: MetaSchema;
  pattern?: SchemaPattern;
  hasGroup?: boolean;
}

export interface ListSchema {
  searchSchema: ISchema;
  tableSchema: ISchema;
  hasCollapsed: boolean;
  searchLogic: LogicListItem[];
  tableLogic: LogicListItem[];
  searchBtnFields: BtnFieldsItem[];
  tableBtnFields: BtnFieldsItem[];
  searchButtons: MetaSchemaData[];
  transformsDone: boolean;
}

export interface FormSchema {
  schema: ISchema;
  btnSchema: ISchema;
  buttons: MetaSchemaData[];
  logicList: LogicListItem[];
  btnFields: BtnFieldsItem[];
  pattern: SchemaPattern;
  transformsDone: boolean;
}

export interface Properties {
  [code: string]: ISchema;
}

export interface PropertiesSchema {
  propertiesSchema: Properties;
  buttons: MetaSchemaData[];
  logicList: LogicListItem[];
  btnFields: BtnFieldsItem[];
  pattern: SchemaPattern;
  transformsDone: boolean;
}

export enum RunSchemaEnum {
  formSchema = 'formSchema',
  propertiesSchema = 'propertiesSchema',
  listSchema = 'listSchema',
}
