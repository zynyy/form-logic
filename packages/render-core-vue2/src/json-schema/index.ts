import { IFieldFactoryProps } from '@formily/core';
import { ISchema, ISchemaTransformerOptions, Schema } from '@formily/json-schema';

import { transformFieldProps } from './transformer';

class JsonSchema extends Schema {
  constructor(json: ISchema) {
    super(json);
    return this.fromJSON(json);
  }

  toFieldProps = (options?: ISchemaTransformerOptions): IFieldFactoryProps<any, any> => {
    return transformFieldProps(this, options);
  };
}

export default JsonSchema;
