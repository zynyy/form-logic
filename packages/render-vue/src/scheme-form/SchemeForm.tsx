import { setValidateLanguage } from '@formily/core';

import { FormProvider } from '@/formily-vue';
import { defineComponent, ref, toRef, watch, watchEffect } from 'vue';
import { Empty, Skeleton } from 'ant-design-vue';

import { useNotifyDevtools } from '@/hooks';
import useCreateSchemaField from '@/hooks/useSchemaField';

import { clone } from '@formily/shared';

import { SchemeFormProps, getSchemeFormProps } from '@/scheme-form/interface';
import { provideSchemeForm } from '@/scheme-form/hooks';

const SchemeForm = defineComponent({
  name: 'SchemeForm',
  inheritAttrs: false,
  props: getSchemeFormProps(),
  setup(props: SchemeFormProps) {
    const {
      getLogicConfig,
      extraLogicParams,
      events,
      pattern,
      loading: propsLoading,
      components,
    } = props;

    const SchemaField = useCreateSchemaField();

    useNotifyDevtools(toRef(props, 'form'));

    watchEffect(() => {
      setValidateLanguage(props.language ?? 'zh-CN');
    });

    let loading = !!propsLoading;

    const schemeFormRef = ref({
      pattern,
      getLogicConfig,
      extraLogicParams,
      events,
    });

    watch(
      () => {
        return {
          pattern: props.pattern,
          getLogicConfig: props.getLogicConfig,
          extraLogicParams: props.extraLogicParams,
          events: props.events,
        };
      },
      (nextSchemeFormRef) => {
        schemeFormRef.value = nextSchemeFormRef;
      },
    );

    provideSchemeForm(schemeFormRef);

    return () => {
      const { schema, form } = props;

      return (
        <Skeleton loading={loading}>
          <FormProvider form={form}>
            <form class={`form-id-${form?.id}`}>
              {schema ? (
                <SchemaField schema={clone(schema)} components={components ?? {}} />
              ) : (
                <Empty description="暂无数据" />
              )}
            </form>
          </FormProvider>
        </Skeleton>
      );
    };
  },
});

export default SchemeForm;
