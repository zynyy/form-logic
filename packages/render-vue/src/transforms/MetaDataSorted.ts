import { MetaSchema, MetaSchemaData, MetaSchemaGroup } from '@/interface';

interface ResultReturn {
  columns: MetaSchemaData[];
  searchColumns: MetaSchemaData[];
  tableColumns: MetaSchemaData[];
  button: MetaSchemaData[];
  searchButton: MetaSchemaData[];
  tableButton: MetaSchemaData[];
}

class MetaDataSorted {
  private readonly metaSchema: MetaSchema;
  private readonly hasGroup: boolean = false;

  columnsArray: MetaSchemaData[] = [];

  searchColumnsArray: MetaSchemaData[] = [];

  tableColumnsArray: MetaSchemaData[] = [];

  buttonsArray: MetaSchemaData[] = [];

  searchButtonsArray: MetaSchemaData[] = [];

  tableButtonsArray: MetaSchemaData[] = [];

  groupsOrder: MetaSchemaGroup[] = [];

  groupButtonsArray: MetaSchemaData[] = [];

  constructor(metaSchema: MetaSchema, hasGroup?: boolean) {
    this.metaSchema = metaSchema;
    this.hasGroup = hasGroup;
  }

  private column = (item: MetaSchemaData) => {
    this.columnsArray.push(item);
  };

  private search_column = (item: MetaSchemaData) => {
    this.searchColumnsArray.push(item);
  };

  private table_column = (item: MetaSchemaData) => {
    this.tableColumnsArray.push(item);
  };

  private button = (item: MetaSchemaData) => {
    const { group } = item;

    if (this.hasGroup && group) {
      this.groupButtonsArray.push(item);
    } else {
      this.buttonsArray.push(item);
    }
  };

  private search_button = (item: MetaSchemaData) => {
    this.searchButtonsArray.push(item);
  };

  private table_button = (item: MetaSchemaData) => {
    this.tableButtonsArray.push(item);
  };

  run = () => {
    const { data, group } = this.metaSchema || {};

    data?.forEach((item) => {
      const { type } = item || {};
      this[type]?.(item);
    });

    const occupyAllCodes =
      group
        ?.filter((item) => {
          const { mode } = item || {};
          return !!mode;
        })
        ?.reduce((acc, cur) => {
          const { modeCodes, code } = cur || {};
          return acc.concat(code).concat(modeCodes);
        }, [])
        .filter((val) => val) || [];

    this.groupsOrder =
      group?.filter((item) => {
        const { mode, code } = item || {};
        return !occupyAllCodes.includes(code) || !!mode;
      }) || [];
  };

  result = (): ResultReturn => {
    this.run();
    return {
      columns: this.columnsArray,
      searchColumns: this.searchColumnsArray,
      tableColumns: this.tableColumnsArray,
      button: this.buttonsArray,
      searchButton: this.searchButtonsArray,
      tableButton: this.tableButtonsArray,
    };
  };
}

export default MetaDataSorted;
