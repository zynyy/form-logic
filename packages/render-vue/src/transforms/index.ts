export * from './interface';

export { default as TransformsSchema } from './TransformsSchema';

