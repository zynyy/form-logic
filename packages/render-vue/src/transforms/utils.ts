import { StrNumBool } from '@/interface';

export const strNumBoolToBoolean = (val?: StrNumBool) => {
  return val === undefined ? false : Boolean(Number(val));
};
