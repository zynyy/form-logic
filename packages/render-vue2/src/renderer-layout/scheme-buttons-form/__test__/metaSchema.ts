import { MetaSchema } from '@/interface';

export default {
  code: 'Api_C',
  labelCol: 6,
  wrapperCol: 18,
  defaultSearchColumn: 6,
  group: [],
  data: [
    {
      type: 'column',
      schemaType: 'string',
      required: '0',
      hidden: '0',
      disabled: '0',
      hiddenName: '0',
      wrap: '0',
      logics: [],
      code: 'url',
      name: '请求接口',
      hasSerialNo: '0',
      hasSort: '0',
    },
    {
      type: 'column',
      schemaType: 'string',
      required: '1',
      hidden: '0',
      disabled: '0',
      hiddenName: '0',
      wrap: '0',
      logics: [],
      code: 'method',
      name: '请求方式',
      defaultValue: 'get',
      component: 'input',
    },
  ],
  model: 'Api',
  codeSuffix: 'C',
  name: '组件接口配置',
} as MetaSchema;
