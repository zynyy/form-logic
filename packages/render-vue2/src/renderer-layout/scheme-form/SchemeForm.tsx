import {
  clone,
  FormProvider,
  isEmpty,
  observer,
  setValidateLanguage,
} from '@formlogic/render-core-vue2';
import { Empty, Form, Skeleton, SkeletonItem } from 'element-ui';
import { get } from 'lodash-es';
import vue, { defineComponent, onMounted, ref, toRef, watch } from 'vue';

import { useNotifyDevtools } from '@/hooks';
import useCreateSchemaField from '@/hooks/useSchemaField';
import { provideSchemeForm } from '@/renderer-layout/scheme-form/hooks';
import { getSchemeFormProps, SchemeFormProps } from '@/renderer-layout/scheme-form/interface';

vue.use(Skeleton).use(SkeletonItem);

const SchemeForm = observer(
  defineComponent({
    name: 'SchemeForm',
    inheritAttrs: false,
    props: getSchemeFormProps(),
    setup(props: SchemeFormProps) {
      const { getLogicConfig, extraLogicParams, pattern, components } = props;

      const SchemaField = useCreateSchemaField();

      useNotifyDevtools(toRef(props, 'form'));

      const setLanguage = () => {
        setValidateLanguage(props.language ?? 'zh-CN');
      };

      watch(
        () => props.language,
        () => {
          setLanguage();
        },
      );

      onMounted(() => {
        setLanguage();
      });

      const schemeFormRef = ref({
        pattern,
        getLogicConfig,
        extraLogicParams,
      });

      watch(
        () => {
          return {
            pattern: props.pattern,
            getLogicConfig: props.getLogicConfig,
            extraLogicParams: props.extraLogicParams,
          };
        },
        (nextSchemeFormRef) => {
          schemeFormRef.value = nextSchemeFormRef;
        },
      );

      provideSchemeForm(schemeFormRef);

      return () => {
        const { schema, form, loading } = props;

        const hasEmpty = isEmpty(get(schema, 'properties.formLayout.properties'));

        const formId = form?.id || 'formId';

        return (
          <Skeleton
            loading={loading}
            animated
            throttle={500}
            style={{
              height: '100%',
            }}
          >
            <FormProvider key={formId} form={form}>
              <Form
                class={`form-id-${formId}`}
                style={{
                  height: '100%',
                }}
              >
                {!hasEmpty ? (
                  <SchemaField schema={clone(schema)} components={components ?? {}} />
                ) : (
                  <Empty description="暂无数据" />
                )}
              </Form>
            </FormProvider>
          </Skeleton>
        );
      };
    },
  }),
);

export default SchemeForm;
