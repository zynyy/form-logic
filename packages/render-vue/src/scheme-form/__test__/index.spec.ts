import { mount } from '@vue/test-utils';

import SchemeForm from '../SchemeForm';
import { createForm, Form } from '@formily/core';
import { TransformsSchema } from '@/transforms';

let form: Form = null;

import metaSchema from './metaSchema';
import { ISchema } from '@formily/json-schema';

let schema: ISchema = null;
describe('scheme-form', () => {
  let wrapper;

  beforeEach(() => {
    form = createForm();

    const formSchema = new TransformsSchema({
      metaSchema,
    }).getFormSchema();

    schema = formSchema.schema;

    wrapper = mount(SchemeForm, {
      props: {
        form,
        schema,
      },
    });
  });

  test(' gen snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
