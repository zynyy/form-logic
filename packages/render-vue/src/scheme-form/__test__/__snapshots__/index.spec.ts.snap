// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scheme-form  gen snapshot 1`] = `
<form class="form-id-snapshot">
  <div style="display: flex; flex-direction: column; gap: 10px;">
    <div style="display: grid; gap: 4px 8px;">
      <div style="grid-row-end: span 1; grid-column: span undefined / auto;" class="form-logic-form-item form-logic-form-item-fullness form-logic-form-item-label-align-right form-logic-form-item-control-align-left form-logic-form-item-control-wrap" data-grid-span="1">
        <div class="form-logic-form-item-label form-logic-form-item-item-col-6">
          <div class="form-logic-form-item-label-content"><span><!----><label>请求接口</label></span></div>
          <!----><span class="form-logic-form-item-colon">:</span>
        </div>
        <div class="form-logic-form-item-control form-logic-form-item-item-col-18">
          <div class="form-logic-form-item-control-content">
            <div class="form-logic-form-item-control-content-component">
              <!----><span><input type="text" class="ant-input"></span>
            </div>
          </div>
          <!---->
        </div>
      </div>
      <div style="grid-row-end: span 1; grid-column: span 1 / auto;" class="form-logic-form-item form-logic-form-item-fullness form-logic-form-item-label-align-right form-logic-form-item-control-align-left form-logic-form-item-control-wrap" data-grid-span="1">
        <div class="form-logic-form-item-label form-logic-form-item-item-col-6">
          <div class="form-logic-form-item-label-content"><span><span class="form-logic-form-item-asterisk">*</span><label>请求方式</label></span></div>
          <!----><span class="form-logic-form-item-colon">:</span>
        </div>
        <div class="form-logic-form-item-control form-logic-form-item-item-col-18">
          <div class="form-logic-form-item-control-content">
            <div class="form-logic-form-item-control-content-component">
              <!----><span><input></span>
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </div>
</form>
`;
